### PR TITLE
[spark] Add tests for buceted scan with partitioned table

### PR DESCRIPTION
**Pull request by ulysses-you**
### Purpose

This pr adds test for buceted scan with partitioned table. Although we do not support bucketed scan with multi-bucket columns, there is a chance to support multi primary keys if  primary keys contains partition column.

### Tests

add test

### API and Format

no

### CODE REVIEW

**Comment by ulysses-you** [comment]
cc @JingsongLi thank you!

**Comment by JingsongLi** [comment]
re-trigger test

File: paimon-spark/paimon-spark-common/src/test/scala/org/apache/paimon/spark/sql/BucketedTableQueryTest.scala [hunk 1]
```diff
@@ -25,7 +25,7 @@ import org.apache.spark.sql.execution.SortExec
 import org.apache.spark.sql.execution.adaptive.AdaptiveSparkPlanHelper
 import org.apache.spark.sql.execution.exchange.ShuffleExchangeLike
 
-class BucketedTableQueryTest extends PaimonSparkTestBase with AdaptiveSparkPlanHelper {
+class c extends PaimonSparkTestBase with AdaptiveSparkPlanHelper {
```
Comment (Zouxxyy) [inline paimon-spark/paimon-spark-common/src/test/scala/org/apache/paimon/spark/sql/BucketedTableQueryTest.scala:28]: Is it a typo?

Comment (ulysses-you) [reply paimon-spark/paimon-spark-common/src/test/scala/org/apache/paimon/spark/sql/BucketedTableQueryTest.scala]: great catch! Not sure how it happened..

**Comment by JingsongLi** [comment]
+1

**Pull request merged**